tools/buildmgr/cbuildgen/config/TIARMCLANG.4.0.1.cmake

```
@@ -0,0 +1,217 @@
+  if(FPU STREQUAL "SP_FPU")
+    set(TIARMCLANG_CPU "-mcpu=cortex-m4 -mfpu=fpv4-sp-d16 -mfloat-abi=hard")
+  else()
+    set(TIARMCLANG_CPU "-mcpu=cortex-m4 -mfloat-abi=soft")
+  endif()
+elseif(CPU STREQUAL "Cortex-M33")
+  if(FPU STREQUAL "SP_FPU")
+    set(TIARMCLANG_CPU "-mcpu=cortex-m33 -mfpu=fpv5-sp-d16 -mfloat-abi=hard")
+  else()
+    set(TIARMCLANG_CPU "-mcpu=cortex-m33 -mfloat-abi=soft")
+  endif()
+elseif(CPU STREQUAL "Cortex-R4")
+  if(FPU STREQUAL "DP_FPU")
+    set(TIARMCLANG_CPU "-mcpu=cortex-r4 -mfpu=vfpv3-d16 -mfloat-abi=hard")
+  else()
+    set(TIARMCLANG_CPU "-mcpu=cortex-r4 -mfloat-abi=soft")
+  endif()
+elseif(CPU STREQUAL "Cortex-R5")
+  if(FPU STREQUAL "DP_FPU")
+    set(TIARMCLANG_CPU "-mcpu=cortex-r5 -mfpu=vfpv3-d16 -mfloat-abi=hard")
+  else()
+    set(TIARMCLANG_CPU "-mcpu=cortex-r5 -mfloat-abi=soft")
+  endif()
+endif()
+if(NOT DEFINED TIARMCLANG_CPU)
+  message(FATAL_ERROR "Error: CPU is not supported!")
+endif()
+
+# Assembler
+
+set(ASM_CPU "${TIARMCLANG_CPU}")
+set(ASM_DEFINES ${DEFINES})
+cbuild_set_defines(ASM ASM_DEFINES)
+
+set(ASM_OPTIONS_FLAGS)
+cbuild_set_options_flags(ASM "${OPTIMIZE}" "${DEBUG}" "${WARNINGS}" "" ASM_OPTIONS_FLAGS)
+
+if(BYTE_ORDER STREQUAL "Little-endian")
+  set(ASM_BYTE_ORDER "-mlittle-endian")
+elseif(BYTE_ORDER STREQUAL "Big-endian")
+  set(ASM_BYTE_ORDER "-mbig-endian")
+endif()
+
+# C Pre-Processor
+
+if(SECURE STREQUAL "Secure" OR SECURE STREQUAL "Secure-only")
+  set(CC_SECURE "-mcmse")
+endif()
+
+set(CPP_FLAGS "-E -P ${TIARMCLANG_CPU} -xc ${CC_SECURE}")
+set(CPP_DEFINES ${LD_SCRIPT_PP_DEFINES})
+cbuild_set_defines(CC CPP_DEFINES)
+if(DEFINED LD_REGIONS AND NOT LD_REGIONS STREQUAL "")
+  set(CPP_INCLUDES "-include \"${LD_REGIONS}\"")
+endif()
+set(CPP_ARGS_LD_SCRIPT "${CPP_FLAGS} ${CPP_DEFINES} ${CPP_INCLUDES} \"${LD_SCRIPT}\" -o \"${LD_SCRIPT_PP}\"")
+separate_arguments(CPP_ARGS_LD_SCRIPT NATIVE_COMMAND ${CPP_ARGS_LD_SCRIPT})
+
+# C Compiler
+
+set(CC_CPU "${TIARMCLANG_CPU}")
+set(CC_DEFINES ${ASM_DEFINES})
+set(CC_BYTE_ORDER ${ASM_BYTE_ORDER})
+set(CC_FLAGS "")
+set(_PI "-include ")
+set(_ISYS "-isystem ")
+set(CC_OPTIONS_FLAGS)
+cbuild_set_options_flags(CC "${OPTIMIZE}" "${DEBUG}" "${WARNINGS}" "${LANGUAGE_CC}" CC_OPTIONS_FLAGS)
+
+if(BRANCHPROT STREQUAL "NO_BRANCHPROT")
+  set(CC_BRANCHPROT "-mbranch-protection=none")
+elseif(BRANCHPROT STREQUAL "BTI")
+  set(CC_BRANCHPROT "-mbranch-protection=bti")
+elseif(BRANCHPROT STREQUAL "BTI_SIGNRET")
+  set(CC_BRANCHPROT "-mbranch-protection=bti+pac-ret")
+endif()
+
+# C++ Compiler
+
+set(CXX_CPU "${CC_CPU}")
+set(CXX_DEFINES "${CC_DEFINES}")
+set(CXX_BYTE_ORDER "${CC_BYTE_ORDER}")
+set(CXX_SECURE "${CC_SECURE}")
+set(CXX_BRANCHPROT "${CC_BRANCHPROT}")
+set(CXX_FLAGS "${CC_FLAGS}")
+set(CXX_OPTIONS_FLAGS)
+cbuild_set_options_flags(CXX "${OPTIMIZE}" "${DEBUG}" "${WARNINGS}" "${LANGUAGE_CXX}" CXX_OPTIONS_FLAGS)
+
+# Linker
+
+set(LD_CPU ${TIARMCLANG_CPU})
+set(_LS)
+
+if(SECURE STREQUAL "Secure")
+  set(LD_SECURE)
```

From the [TI compiler manual](https://software-dl.ti.com/codegen/docs/tiarmclang/compiler_tools_user_guide/compiler_manual/cortex_m_security_extensions/index.html?highlight=implib#building-a-secure-image-and-creating-an-import-library) this seems to be the option for creating the import library, can you please double check?
```suggestion
  set(LD_SECURE "-Wl,--import_cmse_lib_out=\"${OUT_DIR}/${CMSE_LIB}\"")
```